server/routes/trace_analytics_dsl_router.ts

```diff
@@ -48,18 +58,28 @@ export function registerTraceAnalyticsDslRouter(router: IRouter) {
   router.post(
     {
       path: TRACE_ANALYTICS_JAEGER_INDICES_ROUTE,
-      validate: false,
+      validate: {
+        body: schema.any(),
+        query: schema.object({
+          dataSourceMDSId: schema.maybe(schema.string({ defaultValue: '' }))
+        })
+      },
     },
     async (context, request, response) => {
+      const {dataSourceMDSId}  = request.query;
       const params: RequestParams.IndicesExists = {
         index: [JAEGER_INDEX_NAME, JAEGER_SERVICE_INDEX_NAME],
         allow_no_indices: false,
       };
       try {
-        const resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
-          'indices.exists',
-          params
-        );
+        let client;
+        let resp;
+        if (dataSourceEnabled && dataSourceMDSId) {
+          client = context.dataSource.opensearch.legacy.getClient(dataSourceMDSId);
+          resp = await client.callAPI('indices.exists', params);
+        } else {
+          resp = await context.core.opensearch.legacy.client.callAsCurrentUser('indices.exists',params)
+        }
```

Comment:
I see same pattern 3 times. Can we extract it as a function?

Plus, `client` can be const inside if clause.